plutus-tx/src/PlutusTx/List.hs

```diff
@@ -170,10 +170,9 @@ splitAt n xs
   where
     go :: Integer -> [a] -> ([a], [a])
     go _ []     = ([], [])
-    go 1 (y:ys) = ([y], ys)
-    go m (y:ys) = (y:zs, ws)
-      where
-        (zs, ws) = go (Builtins.subtractInteger m 1) ys
+    go m (y:ys)
+      | m `Builtins.equalsInteger` 1 = ([y], ys)
+      | otherwise = let (zs, ws) = go (Builtins.subtractInteger m 1) ys in (y:zs, ws)
```

Comment:
What do you think about making it a `case` instead of a `let`? The `let (..., ...) = ... in (..., ...)` pattern is usually used for extra laziness (for example to [tie the knot](https://wiki.haskell.org/Tying_the_Knot)), since matching in `let` is lazy. Plus, a `case` is more straightforward to read, since the reader doesn't need to account for the possibility of a variable appearing on both the LHS and the RHS of a `let`. Total nitpicking.

Reply:
you can also just add a bang pattern

Reply:
Or that.

Reply:
I was bitten by let assuming irrefutable patterns / lazy patterns before, so `case` I think is better to reason about evaluation